nailgun/entity_mixins.py

```diff
@@ -717,3 +749,99 @@ def create(self, create_missing=None):
 
         """
         return self.read(attrs=self.create_json(create_missing))
+
+
+class EntityUpdateMixin(object):
+    """This mixin provides the ability to update an entity.
+
+    The methods provided by this class work together. The call tree looks
+    like this::
+
+        update → update_json → update_raw → update_payload
+
+    In short, here is what the methods do:
+
+    :meth:`update_payload`
+        Assemble a payload of data that can be encoded and sent to the
+        server.
+    :meth:`update_raw`
+        Make an HTTP PUT request to the server, including the payload.
+    :meth:`update_json`
+        Check the server's response for errors and decode the response.
+    :meth:`update`
+        Create a :class:`nailgun.entity_mixins.Entity` object representing
+        the created entity and populate its fields.
+
+    See the individual methods for more detailed information.
+
+    """
+
+    def update_payload(self, fields=None):
+        """Create a payload of values that can be sent to the server.
+
+        By default, this method behaves just like :func:`_payload`. However,
+        one can also specify a certain set of fields that should be returned.
+        For more information, see :meth:`update`.
+
+        """
+        values = self.get_values()
+        if fields is not None:
+            values = {field: values[field] for field in fields}
```

Comment:
Right. Travis doesn't test 2.6, and I'm not sure how much work it would take to support 2.6. Speaking of Python versions... 3.5 just came out, and that's worth checking out.